### PR TITLE
enhance: [Cherry-pick] pack datacoord Cluster and SessionManager with interface and mock them

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/28861 https://github.com/milvus-io/milvus/issues/28854
pr: https://github.com/milvus-io/milvus/pull/28869

### CODE REVIEW

File: internal/datacoord/session_manager.go [hunk 1]
```diff
@@ -43,20 +46,39 @@ const (
...
 
-// SessionManager provides the grpc interfaces of cluster
-type SessionManager struct {
+type SessionManager interface {
+	AddSession(node *NodeInfo)
+	DeleteSession(node *NodeInfo)
+	GetSessionIDs() []int64
+	GetSessions() []*Session
+
+	Flush(ctx context.Context, nodeID int64, req *datapb.FlushSegmentsRequest)
```
Comment (czs007) [inline internal/datacoord/session_manager.go:55]: how about move all these  to Cluster?

Comment (XuanYang-cn) [reply internal/datacoord/session_manager.go:55]: Good idea, but later

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**